### ISSUE TITLE
Logstash-forwarder outputs log status messages to error logfile

### ISSUE THREAD

**Issue by gebi** [elastic/logstash-forwarder#395]
Hi,

It would be nice if logstash-forwarder would output it's status logfiles to the normal logfile and not the error logfile.

With default config from debian package, `/var/log/logstash-forwarder.log` stays empty, but `/var/log/logstash-forwarder.err` grows with status messages.

**Comment by jordansissel** [comment]
"err" is `stderr`. The name "standard error" has for decades been a false name because there is no consistency in most applications in what goes where (stdout vs stderr).

If you do not want logs going to stderr, you have some options:
* Redirect stderr to stdout
* Set -quiet
* Set -syslog to make logs go to syslog instead

**Comment by gebi** [comment]
I'm using the provided logstash-forwarder init scripts, and it puts status messages in the wrong output file. I'd accept your explanation if there is no .log file, but there is, in this case it should be obvious that status messages about processed events should go the '.log' fie and not to error.

Currently content of logstash-forwarder.err (with logstash-forwarder.log being empty)
```
2015/03/04 09:39:19.012024 Registrar: processing 135 events
2015/03/04 09:41:13.872209 Registrar: processing 110 events
2015/03/04 09:41:58.916064 Registrar: processing 99 events
2015/03/04 09:45:16.401708 Registrar: processing 105 events
```

**Comment by jordansissel** [comment]
@gebi the '.log' is generated by the init script which is generated by pleaserun. An empty .log means nothing is going to stdout and is no cause for alarm, and it being empty is not an indication of an error.

**Comment by jordansissel** [comment]
```
  chroot --userspec "$user":"$group" "$chroot" sh -c "

    cd \"$chdir\"
    exec \"$program\" $args
  " >> /var/log/logstash-forwarder/logstash-forwarder.log 2>> /var/log/logstash-forwarder/logstash-forwarder.err &
```

This is the init script section which does output redirection. logstash-forwarder is not the only program being executed. THe intent of capturing both stdout and stderr is to ensure no meaningful data is lost.

As a workaround, the init script is writable so you can provide your own modifications or replace it entirely.

> to the normal logfile and not the error logfile.

And what, then, if the .err file was empty? In truth, I don't know what full circumstance would result in you not being upset with this project. I'd like to help, but I don't' have enough information to know the path to solution.

**Comment by jordansissel** [comment]
I agree that if your assumption is ".err" means "erorr messages" that you should expect '.log' to have non-error messages. This is not what it means in this context, however.

Given user confusion, I want to fix things, but I'm not super interested in writing to 2 logfiles (logstash doesn't, elasticsearch doesn't, kibana doesn't, etc; these all have 1 log stream per program). So I can offer solutions that I can come up with, but am willing to iterate on other ideas.

* Idea 1: Output all logging to stdout.
* Idea 2: Turn logging off by default
* Idea 3: Set -syslog by default

Regardless, without changes to pleaserun's generator, you'll *always* have the two .err and .log files. What do you want of them? They must exist to capture output in case of failures, so I don't know how to resolve your concerns.

**Comment by jordansissel** [comment]
Idea 4: Change the '.log' file to be '.stdout'. I'll do this in pleaserun shortly.

**Comment by gebi** [comment]
I'd wish for only a single log file called logstash-forwarder.log and a --quiet cmdline option to logstash-forwarder active per default which removes the periodic status messages. Connection attempt outputs are ok (thought only errors would be nicer).

I'd not make syslog the default because of danger of recursive loops. (logstash-forwarder delivering syslog messages to server, generating log messages on connection errors and then try to deliver them to the server). I was burned enough by such things to never ever do this again ;).
That's why logstash-forwarder just logging to a file is a good default.

**Comment by colinmollenhour** [comment]
+1 out of the box if you want to forward /var/log/syslog you generate a lot of noise from lsf itself. Ideally if using syslog there should be two logger instances, one for "info" and one for "error" and the facility should be more specific or specified by the user. Alternatively allow two options for log severities like -info-log=/var/log/... -error-log=syslog.

**Comment by jordansissel** [comment]
> a --quiet cmdline option to logstash-forwarder active per default

@gebi I am trying to have the default settings be appropriate for new users, and many new users often get confused by software that outputs nothing, so I am not convinced that `-quiet` should be the default.

**Comment by jordansissel** [comment]
https://github.com/jordansissel/pleaserun/issues/76 should fix this, I hope. That issue will make the '.log' file become '.stdout', and the current '.err' file will become '.stderr'

**Comment by suyograo** [comment]
Fixed via jordansissel/pleaserun#76
